### PR TITLE
NXP-21383: split repo into core & JSF module for release 8.10

### PR DESCRIPTION
**Pull request by mincong-h**
:warning: Please DON'T merge this PR.

I'm preparing the release 8.10 of Nuxeo Agenda, which aims to split the repo into core and JSF module. I'd like to have some feedback of this commit, in particular:

1. Where should the `build.properties.sample` be placed?
2. Where should the `build.properties` be placed?
3. Your validation about the content of `MANIFEST.MF` in the module _core_.
4. Your validation about the content of `MANIFEST.MF` in the module _jsf_.
5. Your validation about the content of `pom.xml` in the module _core_.
6. Your validation about the content of `pom.xml` in the module _jsf_.

### CODE REVIEW

File: nuxeo-agenda-core/build.properties.sample [hunk 1]
```diff
@@ -0,0 +1,3 @@
+tomcat.dir=/opt/nuxeo-dm
+#mvn.opts=-o
+
```
Comment (mincong-h) [inline nuxeo-agenda-core/build.properties.sample]: Done.

File: nuxeo-agenda-core/build.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  <target name="deploy-tomcat" depends="build" description="Deploy project on a Tomcat instance">
+    <copy todir="${tomcat.plugins.dir}" verbose="true" overwrite="true" flatten="true">
+      <fileset dir="${basedir}">
+        <include name="**/target/*.jar"/>
+        <exclude name="**/target/*-sources.jar"/>
+      </fileset>
+    </copy>
+  </target>
+
+</project>
```
Comment (mincong-h) [inline nuxeo-agenda-core/build.xml]: Done.

File: nuxeo-agenda-core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.nuxeo.ecm.platform</groupId>
+    <artifactId>nuxeo-agenda-parent</artifactId>
+    <version>8.10-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.nuxeo.ecm.platform</groupId>
```
Comment (mincong-h) [inline nuxeo-agenda-core/pom.xml]: Yes, you're right! Done.

File: nuxeo-agenda-jsf/build.properties.sample [hunk 4]
```diff
@@ -0,0 +1,3 @@
+tomcat.dir=/opt/nuxeo-dm
+#mvn.opts=-o
+
```
Comment (mincong-h) [inline nuxeo-agenda-jsf/build.properties.sample]: Done.

File: nuxeo-agenda-jsf/build.xml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  <target name="deploy-tomcat" depends="build" description="Deploy project on a Tomcat instance">
+    <copy todir="${tomcat.plugins.dir}" verbose="true" overwrite="true" flatten="true">
+      <fileset dir="${basedir}">
+        <include name="**/target/*.jar"/>
+        <exclude name="**/target/*-sources.jar"/>
+      </fileset>
+    </copy>
+  </target>
+
+</project>
```
Comment (mincong-h) [inline nuxeo-agenda-jsf/build.xml]: Done.

File: nuxeo-agenda-jsf/pom.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.nuxeo.ecm.platform</groupId>
+    <artifactId>nuxeo-agenda-parent</artifactId>
+    <version>8.10-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.nuxeo.ecm.platform</groupId>
```
Comment (mincong-h) [inline nuxeo-agenda-jsf/pom.xml]: Done.

File: nuxeo-agenda-core/src/main/resources/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -6,10 +6,5 @@ Bundle-Vendor: Nuxeo
 Bundle-Version: 1.0.0
 Nuxeo-Component: OSGI-INF/agenda-core-types-contrib.xml,
  OSGI-INF/agenda-ecm-types-contrib.xml,
- OSGI-INF/agenda-layout-contrib.xml,
  OSGI-INF/agenda-service-framework.xml,
- OSGI-INF/agenda-operations-contrib.xml,
- OSGI-INF/theme-contrib.xml,
- OSGI-INF/agenda-actions-contrib.xml,
- OSGI-INF/agenda-pageproviders-contrib.xml,
```
Comment (mincong-h) [inline nuxeo-agenda-core/src/main/resources/META-INF/MANIFEST.MF]: You're right. I just checked `org.nuxeo.ecm.platform.query.api.PageProviderService`, the class to which component `agenda-pageproviders-contrib.xml` contributes. It 's a backend service providing content to each page of Agenda using NXQL.

Comment (mincong-h) [inline nuxeo-agenda-core/src/main/resources/META-INF/MANIFEST.MF]: Done.

File: nuxeo-agenda-core/pom.xml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-userworkspace-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-relations-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
```
Comment (troger) [inline nuxeo-agenda-core/pom.xml]: Do you really need `servlet-api` for the core module?

File: nuxeo-agenda-jsf/pom.xml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  <artifactId>nuxeo-agenda-jsf</artifactId>
+  <name>Nuxeo Agenda JSF</name>
+  <description>Nuxeo Agenda JSF</description>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-agenda-core</artifactId>
+      <version>8.10-SNAPSHOT</version>
```
Comment (troger) [inline nuxeo-agenda-jsf/pom.xml]: Do not hardcode version here, use `${project.version}` instead.

File: nuxeo-agenda-core/pom.xml (hunk 8 above)
Comment (mincong-h) [inline nuxeo-agenda-core/pom.xml]: No, I don't think so. Thanks for this hint. Actually, I'm analyzing all the possible dependency problems using `mvn dependency:analyze`, suggested by @akervern in slack. I'll update this soon.

**Comment by mincong-h** [comment]
More commits have been added according to your feedback. @akervern, @troger, @jcarsique, could you review this PR again? BTW, they will be squashed into one commit after your validation, before the T&P.

File: pom.xml [hunk 10]
```diff
@@ -7,83 +7,15 @@
   </parent>
 
   <groupId>org.nuxeo.ecm.platform</groupId>
```
Comment (mincong-h) [inline pom.xml]: No, I can't remove this one. Nuxeo Agenda belongs to `org.nuxeo.ecm.platform`, which is different from its parent _nuxeo-addons-parent_. This line is different from the ones in module _core_ or module _jsf_.

File: nuxeo-agenda-jsf/pom.xml (hunk 9 above)
Comment (mincong-h) [inline nuxeo-agenda-jsf/pom.xml]: Done.

**Comment by mincong-h** [comment]
> You are missing a deployment-fragment.xml file in nuxeo-agenda-core/src/main/resources to actually copy the icons to the Nuxeo Server, otherwise nothing will be copied.

I didn't know it before, so I moved it into _JSF_ module. I'll fix it in the next commit. Thx @troger.

BTW, can we add test(s) to prove that icons have been copied to the Nuxeo Server and are available? Would it be too complex to handle?

**Comment by troger** [comment]
Keep them in the core module as they are referenced in the `agenda-ecm-types-contrib.xml` file.

Icons need to be in the core module so that other UIs can access them (such as Web UI / the Mobile application / ...)

**Comment by mincong-h** [comment]
> Keep them in the core module as they are referenced in the agenda-ecm-types-contrib.xml file.

I agree with you.

> Icons need to be in the core module so that other UIs can access them (such as Web UI / the Mobile application / ...)

I agree with you. Could we test how other UIs use these icons?

**Pull request merged**